azure/functions/decorators/function_app.py

```diff
@@ -1938,6 +1968,75 @@ def decorator():
         return wrap
 
 
+class SettingsApi(DecoratorApi, ABC):
+
+    def function_name(self, name: str,
+                      setting_extra_fields: Dict[str, Any] = {},
+                      ) -> Callable[..., Any]:
+        """Set name of the :class:`Function` object.
+
+        :param name: Name of the function.
+        :return: Decorator function.
+        """
+
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_setting(setting=FunctionName(
+                    name=name,
+                    setting_type="setting",
+                    **setting_extra_fields))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def retry(self,
+              strategy: str,
+              max_retry_count: str,
+              delay_interval: Optional[str] = None,
+              minimum_interval: Optional[str] = None,
+              maximum_interval: Optional[str] = None,
+              setting_extra_fields: Dict[str, Any] = {},
+              ) -> Callable[..., Any]:
+        """The retry decorator adds :class:`RetryPolicy` to the function
+        settings object for building :class:`Function` object used in worker
+        function indexing model. This is equivalent to defining RetryPolicy
+        in the function.json which enables function to retry on failure.
+        All optional fields will be given default value by function host when
+        they are parsed by function host.
+
+        Ref: https://aka.ms/azure-function-retry <FIX LINK>
+
+        :param strategy: The retry strategy to use.
+        :param max_retry_count: The maximum number of retry attempts.
+        :param delay_interval: The delay interval between retry attempts.
+        :param minimum_interval: The minimum delay interval between retry
+        attempts.
+        :param maximum_interval: The maximum delay interval between retry
+        attempts.
+        :param setting_extra_fields: Keyword arguments for specifying
+        additional setting fields to include in the setting json.
+        """
+
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_setting(setting=RetryPolicy(
+                    strategy=strategy,
+                    max_retry_count=max_retry_count,
+                    minimum_interval=minimum_interval,
+                    maximum_interval=maximum_interval,
+                    delay_interval=delay_interval,
+                    **setting_extra_fields))
+                return fb
+
+            return decorator()
```

Comment:
Are we planning to add validation to ensure basic combinations are not missed? Better to raise it here

Reply:
No. Validations will be in the host. If we add a validation here it will hide the host error and only show no functions found in the logs

Reply:
We should at least send a warning - better wording of the validation

Reply:
Discussed this offline can validations across the library coming later.